rasa/core/test.py

```diff
@@ -215,9 +223,22 @@ def _clean_entity_results(
     cleaned_entities = []
 
     for r in tuple(entity_results):
-        cleaned_entity = {"text": text}
-        for k in ("start", "end", "entity", "value"):
+        cleaned_entity = {ENTITY_ATTRIBUTE_TEXT: text}
+        for k in (
+            ENTITY_ATTRIBUTE_START,
+            ENTITY_ATTRIBUTE_END,
+            ENTITY_ATTRIBUTE_TYPE,
+            ENTITY_ATTRIBUTE_VALUE,
+        ):
             if k in set(r):
+                if (
+                    k == ENTITY_ATTRIBUTE_VALUE
+                    and EXTRACTOR in set(r)
+                    and r[EXTRACTOR] == "DucklingHTTPExtractor"
+                ):
+                    # convert values from duckling to strings for evaluation as
+                    # target values are all of type string
+                    r[k] = str(r[k])
```

Comment:
Re the conversation we had in slack, i'm not sure if it makes more sense to change it for all extractors (e.g. if someone created a custom RegexExtractor that pulled a number in `int` instead of str)